src/cljs/athens/views/blocks/core.cljs

```diff
@@ -217,34 +231,38 @@
            (swap! state assoc :string/previous string :string/local string))
 
          [:div
-          {:class         ["block-container"
-                           (when (and dragging (not is-selected)) "dragging")
-                           (when is-editing "is-editing")
-                           (when is-selected "is-selected")
-                           (when (and (seq children) open) "show-tree-indicator")
-                           (when (and (false? initial-open) (= uid linked-ref-uid)) "is-linked-ref")]
-           :data-uid      uid
-           :on-drag-over  (fn [e] (block-drag-over e block state))
-           :on-drag-leave (fn [e] (block-drag-leave e block state))
-           :on-drop       (fn [e] (block-drop e block state))}
+          {:class          ["block-container"
+                            (when (and dragging (not is-selected)) "dragging")
+                            (when is-editing "is-editing")
+                            (when is-selected "is-selected")
+                            (when (and (seq children) open) "show-tree-indicator")
+                            (when (and (false? initial-open) (= uid linked-ref-uid)) "is-linked-ref")]
+           :data-uid       uid
+           ;; :show-editable-dom allows us to render the editing elements (like the textarea)
+           ;; even when not editing this block. When true, clicking the block content will pass
+           ;; the clicks down to the underlying textarea. The textarea is expensive to render,
+           ;; so we avoid rendering it when it's not needed.
+           :on-mouse-enter #(swap! state assoc :show-editable-dom true)
+           :on-mouse-leave #(swap! state assoc :show-editable-dom false)
+           :on-click       (fn [e] (doall (.. e stopPropagation) (rf/dispatch [:editing/uid uid])))
+           :on-drag-over   (fn [e] (block-drag-over e block state))
+           :on-drag-leave  (fn [e] (block-drag-leave e block state))
+           :on-drop        (fn [e] (block-drop e block state))}
 
           [presence/presence-popover-info uid {:inline? true}]
 
-          [drop-area-indicator/drop-area-indicator #(when (= drag-target :above) {:opacity "1"})]
+          (when (= (:drag-target @state) :above) [drop-area-indicator/drop-area-indicator {:grid-area "above"}])
 
           [:div.block-body
-           [:button.block-edit-toggle
-            {:on-click (fn [e]
-                         (when (false? (.. e -shiftKey))
-                           (rf/dispatch [:editing/uid uid])))}]
-
-           [toggle/toggle-el uid-sanitized-block state linked-ref]
-           [context-menu/context-menu-el uid-sanitized-block state]
+           (when (seq children)
+             [toggle/toggle-el uid-sanitized-block state linked-ref])
+           (when (:context-menu/show @state)
+             [context-menu/context-menu-el uid-sanitized-block state])
```

Comment:
Do you think it is faster to use the `when` outside of the element rather than inside?

Reply:
I'd assume it’s faster outside because it doesn't have to evaluate anything about the element itself, just the outer `when`, but this isn't my strong suit.